primitives/transaction/src/control_transaction.rs

```diff
@@ -0,0 +1,65 @@
+use std::{error::Error, fmt, ops};
+
+use nimiq_primitives::account::AccountType;
+use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::Transaction;
+
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct ControlTransaction(Transaction);
+
+#[derive(Clone, Debug)]
+pub struct ControlTransactionError(Transaction);
+
+impl fmt::Display for ControlTransactionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        "not a control transaction".fmt(f)
+    }
+}
+
+impl Error for ControlTransactionError {}
+
+impl From<ControlTransactionError> for Transaction {
+    fn from(tx: ControlTransactionError) -> Transaction {
+        tx.0
+    }
+}
+
+impl TryFrom<Transaction> for ControlTransaction {
+    type Error = ControlTransactionError;
+
+    fn try_from(tx: Transaction) -> Result<ControlTransaction, ControlTransactionError> {
+        if tx.sender_type != AccountType::Staking && tx.recipient_type != AccountType::Staking {
```

Comment:
Should we also check if the value is 0 here?

Reply:
We discussed this shortly, and the answer was yes. However, since that is a breaking change, it cannot go into the next (compatible) testnet release.

Reply:
Signaling transactions also have `TransactionFlags::SIGNALING` set.

The parsed `IncomingStakingTransactionData` has a `.is_signaling()` method:
https://github.com/nimiq/core-rs-albatross/blob/21d91b3187e234c3319d9e3d774512fb04eb3c35/primitives/transaction/src/account/staking_contract/structs.rs#L90